src/elisp/treemacs-extensions.el

```diff
@@ -447,28 +449,33 @@ additional keys."
                          ,(format "The handle to update the the %s extension." name))
                        (defun ,ext-name ()
                          (treemacs-with-writable-buffer
-                          (-let [pr (make-treemacs-project
+                          (let ((pr (make-treemacs-project
                                      :name ,root-label
                                      :path ,root-key-form
-                                     :path-status 'extension)]
+                                     :path-status 'extension))
+                                (button-start (point)))
                             (treemacs--set-project-position ,root-key-form (point-marker))
                             (setq-local ,project-var-name pr)
-                            (insert (propertize "a"
+                            (insert (propertize "\n"
```

Comment:
Print a `\n`, since the button has to be on its own line. Otherwise, it will end up on the same line as some other button, and expanding it will insert the contents after the second button on the line.